### DISCUSSION TITLE
404 when trying to load File Browser (filebrowser)

### DISCUSSION THREAD

**Discussion by AlHolden** [MasaCMS/MasaCMS#58] [Q&A]
QUESTION: Which version or setting provides functionality for the 'filebrowser' to load?

Our current autoupdateurl is https://github.com/MasaCMS/MasaCMS/archive/main.zip
I'm running a local development instance @ http://127.0.0.1:8888/Masa/admin/...

When I click the admin nav link - the current console error is a 404:
GET http://127.0.0.1:8888/Masa/index.cfm/_api/json/v1/acd/filebrowser//browse?resourcepath=User_Assets&directory=&settings=1&

" /Masa/index.cfm/_api/json/v1/acd/filebrowser//browse
The origin server did not find a current representation for the target resource or is not willing to disclose that one exists."

FWIW - Permissions granted all over the place. "acd" is the current site with a "127.0.0.1" domain. There is no literal folder called 'filebrowser' under  /sites/acd/ - but you probably knew that. ;-]

Thanks for any help,
Al

Core Version via update: 7.3
App Server Lucee (5.3.8.206)
DB Server MySQL (5.5.5-10.2.40-MariaDB-log)
Java Version 11.0.11
OS Version Linux (5.13.0-30-generic)

This could be related to: https://github.com/MasaCMS/MasaCMS/issues/44

**Comment by sejourla** [comment]
A couple of things/suggestions:

- it appears like you are running Masa docker instance in a sub-folder locally ... not sure why you need it to be in a sub-folder if it's a local instance, but that might be the source of the problem if it's not configured correctly.
- if so, you definitely don't want the folder to be named "Masa" as that will conflict with the "Masa" namespace
- if you are running it in a sub-folder, make sure you have you set the "context" value in the config/settings.ini.cfm, ie. context=/something (where something is the name of the folder that is *not* "Masa" or "m" or "$")

Hope that helps.

**Reply by AlHolden** [reply]
Thanks @sejourla for the help. I followed your advice, and renamed..
/home/......./Lucee/lucee-express-5.3.8.206/webapps/ROOT/Masa
to
/home/......./Lucee/lucee-express-5.3.8.206/webapps/ROOT/masa_cms
changed the context line to the new name. The file manager still does not load (404)
I tried setting a local domain in my hosts file and a <Host> setup in server.xml, but that didn't work either.

To summarize, this is a development laptop with multiple web applications in development under /webapps/ROOT/
They all work fine and are just loaded individually by sub folder. YES. This is a lazy setup, the need to do some Masa stuff is now apparently forcing my lazy hand.

This is the Masa board, not the Lucee board; but I'm having issues finding a step-by-step for setting individual domain/sites using Lucee EXPRESS VERSION. The blogs I've found are all about production Lucee stacks - and the settings just don't line up.

 I would expect a lot more than _just the File Manager_ to break if I had this so wrong, but I'm probably going to need to start with a fresh Lucee/Masa and work backwards to all my other apps from there.

Anyway, thanks for trying,
Al
